src/Common/org/lobobrowser/util/gui/GUITasks.java

```diff
@@ -85,12 +89,30 @@ public static void drawDashed(final Graphics g, int x1, int y1, int x2, int y2,
           g.drawLine(x1, firstY, x1, topY);
         }
       }
+
     } else {
       // Not supported
       g.drawLine(x1, y1, x2, y2);
     }
   }
 
+  public static void drawDotted(Graphics g, int x1, int y1, int x2, int y2, Color color, float width) {
+
+    Graphics2D g2d = (Graphics2D) g;
+    g2d.setColor(color);
+
+    float dot = width * 0.5f;
+    float gap = width * 0.65f;
+
+    float[] dashingPattern1 = { dot, gap };
+    Stroke stroke1 = new BasicStroke(width, BasicStroke.CAP_BUTT,
+        BasicStroke.JOIN_BEVEL, 2.0f, dashingPattern1, 0.0f);
+
+    g2d.setStroke(stroke1);
+    int c = Math.round(width);
+    g2d.drawLine(x1, y1, x2 - c, y2);
```

Comment:
In this function, you will need to either:
1. Create a new graphics object and destroy it, or
2. Save the current stroke using `getStroke` and restore it manually with `setStroke` after drawing the line.